pkg/config/decode.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+
+	"github.com/pelletier/go-toml"
+)
+
+// decode parses a TOML tree (and environment variables) according to field tags set in the Config struct.
+// lookupEnv is used to look-up for env variables, if not provided it defaults to os.LookupEnv.
+// Essentially, decode expects `toml` and `env` field tags to be set. Cf. Config for reference.
```

Comment:
What does `field tags` means?

Reply:
These are metadata one can add to struct fields, they can then be read through the reflect package. This is how we define the `toml` and `env` expected values for each config property.